### ISSUE TITLE
secure.lloydsbank.co.uk: PDF Statement tab won't activate

### ISSUE THREAD

**Issue by Sergey-Lyapin** [AdguardTeam/AdGuardForSafari#879]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I made sure all content blockers are enabled in Safari settings
- [x] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

AdGuard for Safari v1.11.13

### Environment

```markdown
- OS:
- Safari version:
```

### What filters do you have enabled?

AdGuard Base filter, AdGuard Tracking Protection filter

### Issue Details

Steps to reproduce:
1.
2.
3.

### Expected Behavior

PDF Statement tab activates with adguard active

### Actual Behavior

PDF Statement tab won't activate with adguard active

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  ![Screenshot 1](https://reports-img.adguard.com/A7cR1j4.png)
</details>
<details>
  <summary>Screenshot 2</summary>

  ![Screenshot 2](https://reports-img.adguard.com/gpDD07v.png)
</details>

### Additional Information
`@@||lloydsbank.co.uk^$document` or whitelisting `lloydsbank.co.uk` does not help.
This issue is not reproduced in Chrome.

Check out the correspondence in the original issue:
https://github.com/AdguardTeam/AdguardFilters/issues/153992

**Comment by alexx7311** [comment]
@NeilW The issue will be closed in a few days if we don't hear back from you.

**Comment by NeilW** [comment]
Apologies for the delay. 

I have Adguard active in Safari.

<img width="236" alt="Screenshot 2023-09-19 at 10 14 38" src="https://github.com/AdguardTeam/AdGuardForSafari/assets/5795/16bc29e2-f475-4646-bcca-c678a6326177">

with all the Filters turned off the webpage is functional throughout multiple page refreshes (which simulates moving about the site). 

After I switch filter set in AdGuard, I 'Empty Caches' (Option-Command-E) and refresh the page. I then refresh further to see if the PDF selection locks up. 

With just the Base Filter engaged, I get PDF selection lock ups ever three or four refreshes. There doesn't appear to be a consistent pattern, just that it is different from when the filters are entirely off. Hitting refresh a few more times frees it up. 

With just the Protection Tracker engaged, the webpage is functional throughout multiple page refreshes.

With both the Protection Tracker and Base Filter engaged, the PDF selection locks completely on the first refresh. It can be cleared for one refresh by 'Empty caches' again, and then it locks once again on any further refreshes. 

I loaded the Tracker Rules into the user filter and engaged one the Base Filter and the User Filter. I confirmed that this causes the same issue as having Protection Tracker and Base Filter engaged. 

Bisecting the file lead to the same result - on both halves of the file. Running the bisect further showed that the issue was there with 512 rules, started to ease once the CPU died down at 256 rules, and was pretty much the same as the Base Filter at 128 rules. 

That was the same whatever rules from the filter were selected. It seems to be tied to the quantity of rules, not the specific rules themselves. 

Which suggests some sort of length of processing timing issue somewhere when the total ruleset gets much above 40000 rules. 

HTH
